modules/statistics/php/statistics_site.class.inc

```diff
@@ -40,12 +40,24 @@ class Statistics_Site extends \NDB_Menu
      */
     function _hasAccess(\User $user) : bool
     {
-        $centerID = htmlspecialchars($_REQUEST['CenterID']);
         //TODO: Create a permission specific to statistics
-        $accessAllProfiles = $user->hasPermission('access_all_profiles') &&
-            $user->hasPermission('data_entry');
-        return $user->hasCenterPermission('data_entry', $centerID) ||
-            $accessAllProfiles;
+        $hasAccessToAllProfiles = $user->hasAllPermissions(
+            array(
+             'access_all_profiles',
+             'data_entry',
+            )
+        );
+        // If a CenterID is passed in the request, check if the user has the
+        // data_entry permission at the site/center specified by CenterID.
+        $hasCenterPermission = false;
+        if (!empty($_REQUEST['CenterID'] ?? '')) {
```

Comment:
I don't think the null coalesce is necessary here.. empty already handles null (and not set)

Reply:
It does but PHP will complain anyway (give a PHP Notice) if you try to access `$_REQUEST['CenterID']`when 'CenterID' isn't passed

Reply:
I'm pretty sure that's only true outside of an `empty()` call. `empty` has the same properties as null coalesce about silencing unset warnings.

(I just double checked:

```
$ php --version
PHP 7.3.4-2 (cli) (built: Apr 13 2019 19:05:48) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.4, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.4-2, Copyright (c) 1999-2018, by Zend Technologies
$ php -a
Interactive mode enabled

php > ini_set('error_reporting', E_ALL | E_STRICT);
php > echo $foo['bar'];
PHP Notice:  Undefined variable: foo in php shell code on line 1
php > if (empty($foo['bar'])) { echo 'true'; } else { echo 'false'; }
true
php > ^D
```

)

Reply:
Alright I'll update this